### PR TITLE
Add/fix fields in `CREATE MIGRATION`

### PR DESCRIPTION
**Pull request by tailhook**
1. `message` field is now reflected in `schema::Migration`
2. Added `generated_by` field to the migration
3. DDL-generated migrations are now marked with `generated_by := MigrationGeneratedBy.DDLStatement`.

### CODE REVIEW

**Comment by tailhook** [comment]
Also I've noted that you can't do `SET message := ` in `START MIGRATION`:
```
edgedb> start migration to { module default { type X; }};
OK: START MIGRATION
edgedb[tx]> populate migration;
OK: POPULATE MIGRATION
edgedb[tx]> set message := 'from started';
error: EdgeQLSyntaxError: Unexpected 'message'
  ┌─ query:1:5
  │
1 │ set message := 'from started';
  │     ^^^^^^^ error

edgedb[tx:failed]> rollback;
```

This is a bug, right? But I'm not sure if it's fixable, as it conflicts with `set {module|global|alias}` syntax a bit.

**Comment by msullivan** [comment]
> This is a bug, right? But I'm not sure if it's fixable, as it conflicts with `set {module|global|alias}` syntax a bit.

It's probably fixable, in this specific case of `set message`; probably not for arbitrary things?

File: edb/edgeql/parser/grammar/ddl.py [hunk 1]
```diff
@@ -2865,22 +2871,44 @@ def reduce_CreateMigration(self, *kids):
...
+                case 'message':
+                    message = f.value
+                case 'generated_by':
+                    generated_by = f.value
+
         self.val = qlast.CreateMigration(
             name=kids[2].val.name,
             parent=kids[2].val.parent,
             body=kids[3].val.body,
+            message=message,
```
Comment (msullivan) [inline edb/edgeql/parser/grammar/ddl.py:2887]: I think that if we populate the `commands` field here with `kids[3].val.fields`, the generic mechanisms for handling `SET field := ...` should pick up the commands and work without needing to have a `message` or `generated_by` field in `qlast.CreateMigration`, without needing to pick them out in the parser, and thus also without needing to handle them in `_cmd_from_ast`

**Review by msullivan — COMMENTED** [review]
This is all fairly reasonable but I *think* that there is a simpler way that leverages our generic mechanisms for handling `SET`. Try to use that and follow up with me if it doesn't work?

**Comment by tailhook** [comment]
> This is all fairly reasonable but I _think_ that there is a simpler way that leverages our generic mechanisms for handling `SET`. Try to use that and follow up with me if it doesn't work?

It works. Thanks! Feels a bit too magical for someone not familiar enough with the codebase, though.

**Comment by tailhook** [comment]
Oh, by the way, I've just noticed that I've added uniqueness check into the quite generic check. Please double check that it's fine (looks like it's only used in two places, but I'm not sure). Maybe some `SET field` statements should be overridable?

**Comment by msullivan** [comment]
> Oh, by the way, I've just noticed that I've added uniqueness check into the quite generic check. Please double check that it's fine (looks like it's only used in two places, but I'm not sure). Maybe some `SET field` statements should be overridable?

Yeah, that seems right.

**Review by msullivan — APPROVED** [review]
Oh I thought I approved this yesterday!

**Pull request merged**